### ISSUE TITLE
Redis / PostgresQL reward caching; performance comparison

### ISSUE THREAD

**Issue by pstjohn** [NREL/rlmolecule#14]
Would be great to get a quantitative estimation of the performance for redis / postgresQL, perhaps some wrapper functions that make caching easier using either backend. 

I've noticed that sometimes the reward calculation takes a while -- with many simultaneous games, we sometimes see the entry not present in the reward buffer, but it gets added by a different node by the time its complete. so this can throw errors

Other things we could cache depending on performance:
* node values and prior logits (these would have to get cleared every time the model weights are updated)
